src/ContactPeople/ContactPersonForm/EmbeddedContactForm.js

```diff
@@ -0,0 +1,138 @@
+import React, { Component, Fragment } from 'react';
+import PropTypes from 'prop-types';
+import {
+  Field,
+  getFormValues,
+} from 'redux-form';
+import { FormattedMessage } from 'react-intl';
+
+import {
+  Button,
+  Col,
+  Icon,
+  Row,
+} from '@folio/stripes-components';
+
+import css from './AddContactDetailsComponent.css';
+
+class EmbeddedContactForm extends Component {
+  static propTypes = {
+    change: PropTypes.func.isRequired,
+    dispatch: PropTypes.func.isRequired,
+    fieldComponents: PropTypes.object.isRequired,
+    fieldKey: PropTypes.number.isRequired,
+    fieldName: PropTypes.string.isRequired,
+    handleDelete: PropTypes.func.isRequired,
+    labelForFieldsGroup: PropTypes.node.isRequired,
+    labelForPrimaryFieldsGroup: PropTypes.node.isRequired,
+    name: PropTypes.string.isRequired,
+    store: PropTypes.object.isRequired,
+    visibleFields: PropTypes.arrayOf(PropTypes.string).isRequired,
+    requiredFields: PropTypes.arrayOf(PropTypes.string),
+  }
+
+  static defaultProps = {
+    requiredFields: [],
+  };
+
+  setPrimary = (id) => {
+    const { dispatch, change, store, name } = this.props;
+    const formValues = getFormValues('EditContact')(store.getState());
+
+    formValues[name].forEach((a, i) => {
+      if (i === id) {
+        dispatch(change(`${name}[${i}].isPrimary`, true));
+      } else {
+        dispatch(change(`${name}[${i}].isPrimary`, false));
+      }
+    });
+  }
+
```

Comment:
What do you think about 
```suggestion
  setPrimary = (id) => () => {
    const { dispatch, change, store, name } = this.props;
    const formValues = getFormValues('ContactPersonForm')(store.getState());
    const addresses = formValues[name]
      .map((item, index) => (
        index === id
          ? { ...item, isPrimary: true }
          : { ...item, isPrimary: false }));

    dispatch(change(`${name}`, addresses));
  };

  primaryRadio = ({ input, ...props }) => {
    return (
      <label className={css.primaryToggle} htmlFor={props.fieldKey}>
        <input
          type="radio"
          checked={input.value}
          id={props.fieldKey}
          onChange={this.setPrimary(this.props.fieldKey)}
        />
        {this.props.labelForPrimaryFieldsGroup}
      </label>
    );
  };

```